components/infra-proxy-service/server/nodes.go

```diff
@@ -12,14 +12,38 @@ import (
 	"github.com/chef/automate/components/infra-proxy-service/validation"
 )
 
+// GetNodes fetches the nodes from chef infra server
+func (s *Server) GetNodes(ctx context.Context, req *request.Nodes) (*response.Nodes, error) {
+	c, err := s.createClient(ctx, req.OrgId, req.ServerId)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := c.fetchAffectedNodes(ctx, "*:*")
```

Comment:
Fetch node list from chef-server using search `node` index may we can refer: `knife search node "*:*"`